includes/admin/class-rop-admin.php

```diff
@@ -1760,4 +1780,69 @@ public static function rop_check_reached_sharing_limit( $sharing_type = 'tw' ) {
 		}
 		return false;
 	}
+
+	/**
+	 * Check if the user should see the notification about the new X/Twitter limit.
+	 *
+	 * @return bool Whether the user should see the notification.
+	 */
+	public function should_show_twitter_limit_notification() {
+
+		$product_plan  = apply_filters( 'product_rop_license_plan', 0 );
+
+		// Marketer plan should not see the notification. Including lifetime plan.
+		if ( 3 === $product_plan || 6 === $product_plan ) {
+			return false;
+		}
+
+		$service_used = get_option( 'rop_twitter_last_posted', false );
+
+		// Show no notification if the user has not used X/Twitter to share posts.
+		if ( false === $service_used ) {
+			return false;
+		}
+
+		// Get the last date when the notification about new X/Twitter limit was shown.
+		$last_date = get_option( 'rop_twitter_notice_date', false );
+
+		// Show it if it was never shown.
+		if ( $last_date === false ) {
+			return true;
+		}
+
+		$last_date     = strtotime( $last_date );
```

Comment:
`strtotime()` can return `false` hence a validation is needed prior to doing a time difference, another option is to cast the `$last_date` to (int) to ensure type coherence.